From aed61ff438e209985f1827ed37df2bc2090e182e Mon Sep 17 00:00:00 2001
From: Milad Fa <mfarazma@redhat.com>
Date: Fri, 27 Aug 2021 10:19:39 -0400
Subject: [PATCH] PPC/s390: [sparkplug] skip test on unsupported platforms

Port 732f394c5d88862ccb20a79eabd78cc4b054135f

Original Commit Message:

    StaGlobal didn't write the accumulator, but the baseline implementation
    assumed that it could preserve the accumulator by taking the return
    value of the StoreGlobalIC. This almost always worked, except for
    setters on the global object.

    Fix this by marking StaGlobal as clobbering the accumulator, same as
    StaNamedProperty (StaNamedProperty needs to do this anyway to avoid
    inlined setters from needing to create accumulator-preserving frames;
    StaGlobal would have needed the same thing if we'd ever inlined setters
    for it).

    Also, add a new debug scope, EnsureAccumulatorPreservedScope, to the
    baseline compiler, which checks if the accumulator value is preserved
    across non-accumulator-writing bytecodes. This found a (benign) bug with
    ForInPrepare, so fix that too.

R=leszeks@chromium.org, joransiu@ca.ibm.com, junyan@redhat.com, midawson@redhat.com
BUG=
LOG=N

Change-Id: Id8ada05abeb1a9c7e8a16936c35be9d652c4e8b0
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3124529
Reviewed-by: Leszek Swirski <leszeks@chromium.org>
Commit-Queue: Milad Fa <mfarazma@redhat.com>
Cr-Commit-Position: refs/heads/main@{#76537}
---
 test/mjsunit/mjsunit.status | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/mjsunit/mjsunit.status b/test/mjsunit/mjsunit.status
index fd46f466001..56d549d9dd4 100644
--- a/test/mjsunit/mjsunit.status
+++ b/test/mjsunit/mjsunit.status
@@ -1452,6 +1452,7 @@
 # TODO(v8:11421): Port baseline compiler to other architectures.
 ['arch not in (x64, arm64, ia32, arm, mips64el, mipsel, riscv64, loong64)', {
   'baseline/*': [SKIP],
+  'regress/regress-1242306': [SKIP],
 }],
 
 ##############################################################################
-- 
2.35.1

